content-services/content-sources-backend
# Fixes 5038: add uploads search endpoint

## Summary
This PR adds a new endpoint for checking if a file has already been uploaded (i.e. is present as an artifact in Pulp). The search can be done by a `sha256` hash of a file.

## Testing steps
1. Create an upload repo.
2. Get a hash of any rpm file you want to upload.
3. Search uploads for that hash by using the new endpoint:
    `http :8000/api/content-sources/v1.0/repositories/uploads/search "$( ./scripts/header.sh $YOUR_ORG_ID 1111)" sha256:='["$THE_HASH_OF_THE_RPM"]'`
4. Verify that the response contains the hash in the missing array.
5. Upload that rpm to the upload repo and wait for the upload to finish.
6. Repeat step 3 and verify the response has that rpm hash in the found array.
